classes/migration/upgrade/OJSv3_3_0UpgradeMigration.inc.php

```diff
@@ -41,4 +43,114 @@ public function down() {
 			$table->text('setting_value')->nullable()->change();
 		});
 	}
+
+	/**
+	 * @return void
+	 * @bried reset serialized arrays and convert array and objects to JSON for serialization, see pkp/pkp-lib#5772
+	 */
+	private function _settingsAsJSON() {
+
+		// Convert settings where type can be retrieved from schema.json
+		$schemaDAOs = ['SiteDAO', 'AnnouncementDAO', 'AuthorDAO', 'ArticleGalleyDAO', 'JournalDAO', 'EmailTemplateDAO', 'PublicationDAO', 'SubmissionDAO'];
+		$processedTables = [];
+		foreach ($schemaDAOs as $daoName) {
+			$dao = DAORegistry::getDAO($daoName);
+			$schemaService = Services::get('schema');
+			$schema = $schemaService->get($dao->schemaName);
+			$tableName = $dao->settingsTableName;
```

Comment:
There's no common base class or interface specifying that `settingsTableName` needs to exist (same with `schemaName`); I'd almost rather fully code the details out here rather than adding properties that are not fully used in other tables (e.g. `SiteDAO` has `settingsTableName` per the other PR, but it's never used anywhere but in this bit of ephemeral upgrade code).

Reply:
This code will come and go eventually, but there's a tendency to accumulate details in long-standing DAOs etc. that will eventually be misunderstood and cause trouble.